src/aiidalab_qe/app/submission/__init__.py

```diff
@@ -380,18 +338,22 @@ def _get_code_uuid(code):
                     return None
 
         with self.hold_trait_notifications():
-            # Codes
-            self.pw_code.value = _get_code_uuid(codes["pw"])
-            self.dos_code.value = _get_code_uuid(codes["dos"])
-            self.projwfc_code.value = _get_code_uuid(codes["projwfc"])
-
-    def set_pdos_status(self):
-        if "pdos" in self.input_parameters.get("workchain", {}).get("properties", []):
-            self.dos_code.code_select_dropdown.disabled = False
-            self.projwfc_code.code_select_dropdown.disabled = False
-        else:
-            self.dos_code.code_select_dropdown.disabled = True
-            self.projwfc_code.code_select_dropdown.disabled = True
+            for name, code in self.codes.items():
+                code.value = _get_code_uuid(codes.get(name))
+
+    def set_codes_status(self):
+        """Disable code if no related property is selected."""
+        # disable all codes except pw
+        for name, code in self.codes.items():
+            if name == "pw":
+                continue
+            code.code_select_dropdown.disabled = True
+        # enable code if the related property is selected.
+        properties = self.input_parameters.get("workchain", {}).get("properties", [])
+        for identifer, codes in self.code_entries.items():
```

Comment:
```suggestion
        # Update the layout of code widgets
        for identifer, codes in self.code_entries.items():
```
Another nitpick, possible to simplify this for..if..for ?

Reply:
I removed the `if`.